File: sdk/src/main/java/org/maibot/sdk/ioc/Specify.java
package org.maibot.sdk.ioc;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 用于在构造方法参数上指定注入的实现类
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.PARAMETER)
public @interface Specify {
    String name();
}

File: sdk/src/main/java/org/maibot/sdk/ioc/Specify.kt
package org.maibot.sdk.ioc

/**
 * 当用在构造方法参数上时，指定注入的实现类名称
 *
 * 当用在类上时，指定该类的名称
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.CLASS)
annotation class Specify(val name: String)

